resources/lib/player/players.py

```diff
@@ -144,18 +144,28 @@ def _set_external_ids(self, tmdb_type, tmdb_id, season, episode, details, requir
             details.set_details(details=self.details_ext_ids, reverse=True)
         return set_detailed_item(tmdb_type, tmdb_id, season, episode, details=details) or {}
 
-    def _set_provider_priority(self):
-        try:  # Check if players are listed in providers and bump priority if found
-            providers = self.details.infoproperties['providers']
-            providers = providers.split(' / ')
-            for k, v in self.players.items():
-                if 'provider' not in v or v['provider'] not in providers:
-                    v['priority'] = v.get('priority', PLAYERS_PRIORITY) + 100  # Increase priority baseline by 100 to prevent other players displaying above providers
-                    continue
-                v['priority'] = providers.index(v['provider']) + 1  # Add 1 because sorted() puts 0 index last
-                v['is_provider'] = True
-        except (KeyError, AttributeError):
-            pass
+    def _get_prioritised_players(self):
+        try:
+            providers = self.details.infoproperties.get('providers', '')
+        except AttributeError:
+            providers = ''
+        providers = tuple(providers.split(' / '))
+
+        def _set_priority(item, _providers=providers):
+            file, player = item
+            player_provider = player.get('provider')
+            if player_provider in _providers:
+                priority = _providers.index(player_provider) + 1  # Add 1 because sorted() puts 0 index last
+                player['is_provider'] = True
+            else:
+                if player.get('is_provider', True):
+                    priority = player.get('priority', PLAYERS_PRIORITY) + 100  # Increase priority baseline by 100 to prevent other players displaying above providers
```

Comment:
Shouldn't this be `if not player.get('is_provider', True)` ?

If the player is **not** a provider then we want to bump its priority value by 100 to make sure it appears further down the list.

Reply:
First a bit of context. I am currently just using this by creating plugin urls, in which case a new instance of the Players class is created everytime it is needed. In this scenario the sorting is only done once in the life of the Players instance.

However I was thinking about potentially importing and using the players module to create a longer lived Players instance.

In this scenario, the sorting may need to be done more than once (everytime a new listitem was required to be played using the Players instance).

If the sorting is done more than once, `not player.get('is_provider')` means that the priority gets bumped everytime. However it only needs to have it's priority changed if it has never been sorted before `('is_provider' not in player)` or if it has been sorted and was previously a priority provider player `(player['is_provider'] == True)`. If it has been sorted and already had the priority changed `(player['is_provider'] == False)` then the priority shouldn't get bumped down again.

That is what the `player.get('is_provider', True)` condition does - it checks for `is_provider` being True (already sorted and previously increased priority) or not present (first time being sorted).

Reply:
Ah okay gotcha. 

I was only thinking in terms of single use since it has to be that way for plugin instances, so I couldn't quite figure out the purpose of writing the condition backwards. Makes way more sense now with the context of reusing and resorting.